### PR TITLE
Feature/retzam/add fetch post

### PR DESCRIPTION
**Pull request by Younggregs**

### CODE REVIEW

File: functions/controllers/post/getPost.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+const _ = require('lodash');
+const HttpStatus = require('http-status-codes');
+const { message, status } = require('../../util/constants');
+
+const { BAD_REQUEST, OK, INTERNAL_SERVER_ERROR } = HttpStatus;
```
Comment (SapeleD3) [inline functions/controllers/post/getPost.js]: const { BAD_REQUEST, OK, INTERNAL_SERVER_ERROR } = require('http-status-codes');
preferred approach now

File: functions/controllers/post/newPost.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+const HttpStatus = require('http-status-codes');
+// const { validateCode } = require('../../util/validator');
+const { status, message } = require('../../util/constants');
+
+const { error, success } = status;
+const {
+    somethingWentWrong,
+} = message;
+const {
+    INTERNAL_SERVER_ERROR,
```
Comment (SapeleD3) [inline functions/controllers/post/newPost.js]: const { BAD_REQUEST, OK, INTERNAL_SERVER_ERROR } = require('http-status-codes');
preferred approach now

**Review by SapeleD3 — COMMENTED** [review]
const user = await db
        .collection(users)
        .where(userId, '==', decodedToken.user_id)
        .limit(1)
        .get();
    req.user.userName = user.docs[0].data().userName;
    req.user.imageUrl = user.docs[0].data().imageUrl;


i have already assigned the user details to req.user
and required data in userName and image to req.user.userName and req.user.imageUrl .. so the create relationship in the post just add those as foriegn key

use this approach to add details of the user tha made a post or added a topic 
const newPost || newTopic = {
      body: req.body.body,
      userName: req.user.userName,
      userImage: req.user.imageUrl,
      createdAt: new Date().toISOString(),
      likeCount: 0,
      commentCount: 0
    };

File: functions/controllers/post/newPost.js (hunk 2 above)
Comment (SapeleD3) [reply functions/controllers/post/newPost.js]: fixing the http status code import would fix the conflict errors..as they have been implemented on my end on master branch

**Review by EngrEric — CHANGES_REQUESTED** [review]
looks good, fix the minor changes and resolve the conflict

**Pull request merged**